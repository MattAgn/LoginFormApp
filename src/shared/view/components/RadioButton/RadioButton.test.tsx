import { screen, userEvent } from '@testing-library/react-native';

import { renderWithProviders } from '#testing/render';

import { RadioButton } from './RadioButton';

describe('RadioButton Component', () => {
  it(`renders the label
    when label is sent`, () => {
    const mockOnPress = jest.fn();

    renderWithProviders(
      <RadioButton
        label="test"
        isChecked={false}
        isDisabled={false}
        onPress={mockOnPress}
      />,
    );

    expect(screen.getByText('test')).toBeOnTheScreen();
    expect(screen).toMatchComponentSnapshot();
  });

  it(`renders of the Checked icon when isChecked is true`, () => {
    const mockOnPress = jest.fn();

    renderWithProviders(
      <RadioButton
        label="test"
        isChecked={true}
        isDisabled={false}
        onPress={mockOnPress}
      />,
    );

    expect(screen.getByTestId('CheckedRadioButton-icon')).toBeOnTheScreen();
    expect(screen).toMatchComponentSnapshot();
  });

  it(`renders the Unchecked icon when isChecked is false`, () => {
    const mockOnPress = jest.fn();

    renderWithProviders(
      <RadioButton
        label="test"
        isChecked={false}
        isDisabled={false}
        onPress={mockOnPress}
      />,
    );

    expect(screen.getByTestId('UnCheckedRadioButton-icon')).toBeOnTheScreen();
    expect(screen).toMatchComponentSnapshot();
  });

  it(`renders the Disabled icon when isDisabled is true`, () => {
    const mockOnPress = jest.fn();

    renderWithProviders(
      <RadioButton
        label="test"
        isChecked={false}
        isDisabled={true}
        onPress={mockOnPress}
      />,
    );

    expect(screen.getByTestId('DisabledRadioButton-icon')).toBeOnTheScreen();
    expect(screen).toMatchComponentSnapshot();
  });

  it(`calls onPress when radio button is pressed`, async () => {
    const mockOnPress = jest.fn();

    renderWithProviders(
      <RadioButton
        label="test"
        isChecked={true}
        isDisabled={false}
        onPress={mockOnPress}
      />,
    );

    await userEvent.press(screen.getByTestId('CheckedRadioButton-icon'));

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it(`does not call onPress when radio button is pressed and isDisabled is true`, async () => {
    const mockOnPress = jest.fn();

    renderWithProviders(
      <RadioButton
        label="test"
        isChecked={true}
        isDisabled={true}
        onPress={mockOnPress}
      />,
    );

    await userEvent.press(screen.getByTestId('DisabledRadioButton-icon'));

    expect(mockOnPress).toHaveBeenCalledTimes(0);
  });
});
