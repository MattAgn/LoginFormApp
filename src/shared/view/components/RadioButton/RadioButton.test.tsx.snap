// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component renders of the Checked icon when isChecked is true 1`] = `"test"`;

exports[`RadioButton Component renders of the Checked icon when isChecked is true 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="test"
    accessibilityRole="radio"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": true,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignSelf": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
          "flex": 0,
          "height": 24,
          "width": 24,
        }
      }
      testID="CheckedRadioButton-icon"
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGCircle
          cx="12"
          cy="12"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
            ]
          }
          r="8"
          stroke={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          strokeWidth="8"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        {
          "width": 8,
        }
      }
    />
    <View>
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#000000",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`RadioButton Component renders the Disabled icon when isDisabled is true 1`] = `"test"`;

exports[`RadioButton Component renders the Disabled icon when isDisabled is true 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="test"
    accessibilityRole="radio"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": false,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignSelf": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
          "flex": 0,
          "height": 24,
          "width": 24,
        }
      }
      testID="DisabledRadioButton-icon"
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12Z"
          fill={
            {
              "payload": 4294111986,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
              "stroke",
            ]
          }
          stroke={
            {
              "payload": 4294111986,
              "type": 0,
            }
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        {
          "width": 8,
        }
      }
    />
    <View>
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#737373",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`RadioButton Component renders the Unchecked icon when isChecked is false 1`] = `"test"`;

exports[`RadioButton Component renders the Unchecked icon when isChecked is false 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="test"
    accessibilityRole="radio"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": false,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignSelf": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
          "flex": 0,
          "height": 24,
          "width": 24,
        }
      }
      testID="UnCheckedRadioButton-icon"
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17317C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8079C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 10.6868 21.7413 9.38642 21.2388 8.17317C20.7363 6.95991 19.9997 5.85752 19.0711 4.92893C18.1425 4.00035 17.0401 3.26375 15.8268 2.7612C14.6136 2.25866 13.3132 2 12 2ZM12 20C10.4178 20 8.87104 19.5308 7.55544 18.6518C6.23985 17.7727 5.21447 16.5233 4.60897 15.0615C4.00347 13.5997 3.84504 11.9911 4.15372 10.4393C4.4624 8.88743 5.22433 7.46197 6.34315 6.34315C7.46197 5.22433 8.88743 4.4624 10.4393 4.15372C11.9911 3.84504 13.5997 4.00346 15.0615 4.60896C16.5233 5.21447 17.7727 6.23984 18.6518 7.55544C19.5308 8.87103 20 10.4177 20 12C20 14.1217 19.1572 16.1566 17.6569 17.6569C16.1566 19.1571 14.1217 20 12 20Z"
          fill={
            {
              "payload": 4291611852,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        {
          "width": 8,
        }
      }
    />
    <View>
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#000000",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`RadioButton Component renders the label
    when label is sent 1`] = `"test"`;

exports[`RadioButton Component renders the label
    when label is sent 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="test"
    accessibilityRole="radio"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": false,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignSelf": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
          "flex": 0,
          "height": 24,
          "width": 24,
        }
      }
      testID="UnCheckedRadioButton-icon"
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M12 2C10.0222 2 8.08879 2.58649 6.4443 3.6853C4.79981 4.78412 3.51809 6.3459 2.76121 8.17317C2.00433 10.0004 1.8063 12.0111 2.19215 13.9509C2.578 15.8907 3.53041 17.6725 4.92894 19.0711C6.32746 20.4696 8.10929 21.422 10.0491 21.8079C11.9889 22.1937 13.9996 21.9957 15.8268 21.2388C17.6541 20.4819 19.2159 19.2002 20.3147 17.5557C21.4135 15.9112 22 13.9778 22 12C22 10.6868 21.7413 9.38642 21.2388 8.17317C20.7363 6.95991 19.9997 5.85752 19.0711 4.92893C18.1425 4.00035 17.0401 3.26375 15.8268 2.7612C14.6136 2.25866 13.3132 2 12 2ZM12 20C10.4178 20 8.87104 19.5308 7.55544 18.6518C6.23985 17.7727 5.21447 16.5233 4.60897 15.0615C4.00347 13.5997 3.84504 11.9911 4.15372 10.4393C4.4624 8.88743 5.22433 7.46197 6.34315 6.34315C7.46197 5.22433 8.88743 4.4624 10.4393 4.15372C11.9911 3.84504 13.5997 4.00346 15.0615 4.60896C16.5233 5.21447 17.7727 6.23984 18.6518 7.55544C19.5308 8.87103 20 10.4177 20 12C20 14.1217 19.1572 16.1566 17.6569 17.6569C16.1566 19.1571 14.1217 20 12 20Z"
          fill={
            {
              "payload": 4291611852,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        {
          "width": 8,
        }
      }
    />
    <View>
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#000000",
            "fontFamily": "OpenSans-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        test
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
